### PR TITLE
solve #306, and bin in $PATH make sense now

### PR DESCRIPTION
**Pull request by zhiyelee**
solve #306, and bin in $PATH make sense now  #196
Almost revert #173 which lead to the bug in #306

Also remove deprecated `customFds` option in spawn

### CODE REVIEW

**Comment by SomeKittens** [comment]
Looks like this fixes more than one edge case.  Can we get the tests passing?

File: index.js [hunk 1]
```diff
@@ -460,13 +461,26 @@ Command.prototype.executeSubCommand = function(argv, args, unknown) {
...
+    }
+  } catch (e) {}
 
   // run it
   args = args.slice(1);
-  args.unshift(local);
-  var proc = spawn('node', args, { stdio: 'inherit', customFds: [0, 1, 2] });
+
+  var proc;
+  if (process.platform !== 'win32') {
```
Comment (zhiyelee) [inline index.js]: After the commit, `bin` file must have shebang itself

**Comment by zhiyelee** [comment]
Closed for too many commits. I'll  open another PR
